### PR TITLE
SONARIAC-943 ARM Bicep support: add support for typedLambdaExpression

### PR DESCRIPTION
**Pull request by petertrr**
Syntactic examples (as there are too few in the docs) can be found [here](https://github.com/Azure/bicep/blob/9957a439da41a8860ff5cfd1f0f77a84f8e1a4d5/src/Bicep.Core.Samples/Files/Functions_LF/main.bicep)

### CODE REVIEW

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepLexicalConstant.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ public class BicepLexicalConstant {
   public static final String NULL = "null";
   public static final String QUOTED_STRING_LITERAL_NO_QUOTES = "(?:(?!'|\\$\\{)[\\s\\S])*+";
   public static final String IDENTIFIER_LITERAL = "[a-zA-Z_][a-zA-Z_0-9]*+";
-  public static final String AMBIENT_TYPE = "(?:array|bool|int|object|string)(?![\\S])";
+  public static final String AMBIENT_TYPE = "(?:array|bool|int|object|string)(?=\\s|\\)|,|\\z)";
```
Comment (petertrr) [inline iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepLexicalConstant.java:45]: Related: https://github.com/SonarSource/sonar-iac/pull/798#discussion_r1259267343

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/TreeFactory.java [hunk 2]
```diff
@@ -272,6 +278,24 @@ public MultilineString multilineString(SyntaxToken openingTripleApostrophe, Synt
...
+  public TypedLocalVariable typedLocalVariable(Identifier identifier, AmbientTypeReference primaryTypeExpression) {
+    return new TypedLocalVariableImpl(identifier, primaryTypeExpression);
+  }
+
+  public TypedVariableBlock typedVariableBlock(SyntaxToken lParen, Optional<SeparatedList<TypedLocalVariable, SyntaxToken>> variableList, SyntaxToken rParen) {
+    return new TypedVariableBlockImpl(lParen, variableList.or(emptySeparatedList()), rParen);
+  }
+
+  public TypedLambdaExpression typedLambdaExpression(TypedVariableBlock typedVariableBlock, AmbientTypeReference primaryTypeExpression, SyntaxToken doubleArrow,
+    Expression expression) {
```
Comment (mstachniuk) [inline iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/TreeFactory.java:290]: Nitpik, consider put all arguments in separate lines, for better readability (long line)

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/api/bicep/TypedLambdaExpression.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.api.bicep;
+
+import org.sonar.iac.arm.tree.api.ArmTree;
+
+public interface TypedLambdaExpression extends ArmTree {
+  @Override
```
Comment (mstachniuk) [inline iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/api/bicep/TypedLambdaExpression.java:25]: I like this idea

**Review by mstachniuk — APPROVED** [review]
LGTM

**Pull request merged**